File: pset1/psets/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>


//--------------------------------------------------------------------------
//scale: 1$ = 100 cents.
int main(void)
{

    int amount = 0;
    int qQuarters = 0, qDimes = 0, qNickels = 0, qPennies = 0, coinUsed = 0;

    //prompt user for an amount of change
    do
    {
        amount = get_float("Your amount ? ");
    }

    //conditions
    while (amount <=  0);

//--------------------------------------------------------------------------

    amount = amount * 10000;
    amount = ceil(amount);

    float x = amount;

    //quarters quantity
    while (x - 2500 >= 0)
    {

        x -= 2500;
        qQuarters++;

    }


    //dimes quantity
    while (x - 1000 >= 0)
    {

        x -= 1000;
        qDimes++;

    }


    //nickels quantity
    while (x - 500 >= 0)
    {

        x -= 500;
        qNickels++;

    }


    //pennies quantity
    while (x - 100 >= 0)
    {

        x -= 100;
        qPennies++;

    }



    coinUsed = qQuarters + qDimes + qNickels + qPennies;

    //print numbers of coins used
    printf("%i\n", coinUsed);

}

File: pset6/cash/cash.py
from cs50 import get_float


# prompt the user for a non negative float
while True:
    change = get_float("Your amount ? ")

    if change >= 0:
        break

change = round(change * 100)

quarters = 0
total = 0
nickels = 0

remainder = change

if change >= 25:
    quarters = int(change / 25)
    remainder = int(change % 25)
    total += quarters

if remainder >= 10:
    dimes = int(remainder / 10)
    remainder = int(remainder % 10)
    total += dimes


if remainder >= 5:
    nickels = int(remainder / 5)
    remainder = int(remainder % 5)
    total += nickels

if remainder >= 0:
    pennies = int(remainder / 1)
    remainder = int(remainder % 1)
    total += pennies

print(total)
